base/array.jl

```diff
@@ -396,11 +404,17 @@ julia> getindex(Int8, 1, 2, 3)
  3
 ```
 """
-function getindex(::Type{T}, vals...) where T
+getindex(::Type{T}, vals...) where T = _getindex_impl(T, vals)
+
+# safe version
+getindex(::Type{T}, vals::T...) where T = (@_safe_getindex_meta; _getindex_impl(T, vals))
+
+function _getindex_impl(::Type{T}, vals) where T
+    @inline
     a = Vector{T}(undef, length(vals))
     if vals isa NTuple
         @inbounds for i in 1:length(vals)
-            a[i] = vals[i]
+            a[i] = getfield(vals, i)
```

Comment:
are you intentionally disabling `inbounds` here, or should this be:
```suggestion
        for i in 1:length(vals)
             a[i] = getfield(false, vals, i)
```